### ISSUE TITLE
Builds for arm64 hosts

### ISSUE THREAD

**Issue by rgov** [bootlin/toolchains-builder#54]
With the availability of ARM-based Macs, Chromebooks, Raspberry Pis, etc., it would be useful to have the toolchains released for arm64 hosts.

**Comment by jarkkojs** [comment]
Asahi Linux is starting to super-solid distribution for M1/M2/M3 Macs. Also Lenovo has been pushing Snapdragon powered laptops for two generations now already.

**Comment by jarkkojs** [comment]
And when compiling systemd-enabled BuildRoot image, I've never been successful using self-bootstrapped toolchain. I.e. sometimes this can be a real functional blocker..

**Comment by tpetazzoni** [comment]
@jarkkojs could you clarify which issues you've encountered when trying to use self-bootstrapped toolchains with Buildroot? Because if that doesn't work on ARM64, toolchains-builder won't be able to build toolchains for ARM64 hosts... as toolchains-builder uses Buildroot to build its toolchains.

**Comment by tpetazzoni** [comment]
@rgov the problem is that in order to provide this, we would need access to ARM64 hosts machines that can run gitlab-runner, and be ready to build ~200 toolchains (it takes quite some time)

**Comment by rgov** [comment]
Arm64 runners are [publicly available](https://github.blog/changelog/2024-06-03-actions-arm-based-linux-and-windows-runners-are-now-in-public-beta/) for GitHub Actions as of June 2024, but understandably this would be a significant change to your CI system.

Could you not use your own cross-compile toolchains to target Arm64 binaries on an x86 system, with qemu-user-static / binfmt_misc available to run any Arm64 binaries that are needed for the build process?

**Comment by p-linnane** [comment]
@rgov Pointed me to this thread when I mentioned having an M1 Mac mini gathering dust. @tpetazzoni I'm happy to volunteer said device as a runner for Bootlin. Just let me know if you're interested.

**Comment by tpetazzoni** [comment]
Hello @p-linnane, interesting! Could you perhaps drop me an e-mail so that we can see how this could work? You can reach me at thomas.petazzoni AT bootlin.com.

**Comment by jarkkojs** [comment]
@tpetazzoni I can **after** next week I'm on holiday for this week still and try to stay off the grid ;-) But yes, I can provide an example for sure.

**Comment by tpetazzoni** [comment]
@rgov @jarkkojs you confirm that here we are talking about toolchains for Linux ARM64 hosts, not MacOS X ARM64 hosts, correct ?

**Comment by rgov** [comment]
> you confirm that here we are talking about toolchains for Linux ARM64 hosts, not MacOS X ARM64 hosts, correct ?

Correct. When I filed the bug I was using the toolchain within a Linux VM on Arm64 hardware while working with the Nvidia Jetson SDK.
